xarray/conventions.py

```diff
@@ -470,6 +476,27 @@ def __getitem__(self, key):
         return values
 
 
+def serialize_char_to_string_array(array):
+    header, frames = distributed.protocol.serialize(array.array)
+    header['__char_to_string_array__array_type'] = header.get('type')
+    return header, frames
+
+
+def deserialize_char_to_string_array(header, frames):
+    type_ = header.pop('__char_to_string_array__array_type', None)
+    if type_ is not None:
+        header['type'] = type_
+    array = distributed.protocol.deserialize(header, frames)
+    return CharToStringArray(array)
```

Comment:
That would be fine, but these classes (other than `LazilyIndexedArray`) are not going to be layered multiple times. So I wouldn't worry too much about that until we settle on a general solution.